lib/rmt/mirror.rb

```diff
@@ -171,7 +179,15 @@ def mirror_packages(metadata_files, repository_dir, repository_url)
 
     raise _('Failed to download %{failed_count} files') % { failed_count: failed_downloads.size } unless failed_downloads.empty?
   rescue StandardError => e
-    raise RMT::Mirror::Exception.new(_('Error while mirroring data: %{error}') % { error: e.message })
+    message ||= e.message
```

Comment:
Why is this line needed? Can the message get set from somewhere else?

Reply:
No, it can not. It is needed to get the error message that originated the exception

Reply:
I was asking why this is `||=` instead of `=`. In which case would `message` already be set? Same in line 127.

Reply:
It turned out that the `||=` was hiding the current exceptions. In the tests this was hiding missing VCRs for example. 
I propose to go a different way and implement the retry login ion the `Downloader` class, where it feels better suited for me. I started to implement that here: https://github.com/SUSE/rmt/pull/907/